bears/general/URLBear.py

```diff
@@ -76,19 +83,19 @@ class URLBear(LocalBear):
 
     @classmethod
     def check_prerequisites(cls):
-        code = cls.get_status_code(
+        head_resp = cls.get_head_response(
             cls.check_connection_url, cls.DEFAULT_TIMEOUT)
         return ('You are not connected to the internet.'
-                if code is None else True)
+                if isinstance(head_resp, Exception) else True)
 
     @staticmethod
-    def get_status_code(url, timeout):
+    def get_head_response(url, timeout):
         try:
-            code = requests.head(url, allow_redirects=False,
-                                 timeout=timeout).status_code
-            return code
-        except requests.exceptions.RequestException:
-            pass
+            head_resp = requests.head(url, allow_redirects=False,
+                                      timeout=timeout)
+            return head_resp
+        except Exception as exc:
```

Comment:
why not RequestException?

Also I think the catching of this exception should ideally be outside, having different return types in different cases is usually bad

Reply:
updated